shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -517,6 +534,13 @@ - (void)updateDisplayConfig {
   return _bridge;
 }
 
+- (nonnull const char*)executableName {
+  // We hold onto the executable name as a static since this value will not change over the process
+  // lifetime and because argv contents are assumed to be stable over the life of the Dart runtime.
```

Comment:
The PR description says:
> Since the string pointers in command_line_argv are required to
> be valid for the lifetime of the Dart runtime

but the embedder.h docs say:

```
  /// The command line arguments used to initialize the project. The strings can
  /// be collected after the call to `FlutterEngineRun` returns.
```

If embedder.cc isn't keeping this around forever, that's a bug in embedder.cc.

At which point this would just be a cache to avoid a small amount of work in the relatively rare case of starting multiple engine instances, which seems needlessly complex.

Reply:
embedder.cc does perform a copy. This API should probably return a string that maintains a copy of the data backing the string (std::string or NSString) instead of just a const string. Once it is given the embedder, it may be collected.

Reply:
Thanks; my initial assumption was we needed to keep it around, but as part of the refactor into two patches (the embedder one and this one) I did end up remembering the copying we do as part of the fml commandline parsing, but somehow the cleanup completely slipped my mind during the refactoring. Thanks for catching this!

We initialize and run further down in the same scope, so we should be alright in terms of pointer lifetime.